### PR TITLE
Send correct hostname to google analytics

### PR DESCRIPTION
**Pull request by andrewrlee**
Fix case when single url is used

### Context

https://trello.com/c/j68vCqlK

When we are using the single prison agnostic url we still want to be able to track per establishment usage via google analytics.

### Intent

This builds up the url based on the session derived establishment id and the single host url. 

### Considerations

URL pattern differs between dev/staging and prod: `-` vs `.` separated.

This doesn't take that into account so will start to report different urls in dev and staging.
I don't think this matters but need to check

### Checklist

- [ ] This PR contains **only** changes related to the above card
- [ ] Tests have been added/updated to cover the change
- [ ] Documentation has been updated where appropriate
- [ ] Tested in Development

### CODE REVIEW

File: helm_deploy/prisoner-content-hub-frontend/templates/_envs.tpl [hunk 1]
```diff
@@ -93,5 +93,9 @@ env:
         secretKeyRef:
           name: {{ include "prisoner-content-hub-frontend.fullname" . }}
           key: azureAdClientSecret
-
+      
+    - name: SINGLE_HOST_NAME
```
Comment (andrewrlee) [inline helm_deploy/prisoner-content-hub-frontend/templates/_envs.tpl:97]: make single hostname available as env var

File: server/config.js [hunk 2]
```diff
@@ -30,11 +23,10 @@ module.exports = {
     gitDate: getEnv('GIT_DATE', '2020-06-21 12:12:12'),
   },
   cookieSecret: getEnv('COOKIE_SECRET', 'keyboard cat'),
+  singleHostName: getRequiredEnv('SINGLE_HOST_NAME', 'localhost'),
   auth: {
-    clientId: getEnv('AZURE_AD_CLIENT_ID', { requireInProduction: true }),
-    clientSecret: getEnv('AZURE_AD_CLIENT_SECRET', {
-      requireInProduction: true,
-    }),
```
Comment (andrewrlee) [inline server/config.js:37]: these were broken as they required a default var so used the ` { requireIn...` as the default value.
Extracted out a `getRequiredEnv ` instead and provided noddy defaults

File: server/routes/analytics.js [hunk 3]
```diff
@@ -20,10 +21,11 @@ const createAnalyticsRouter = ({ analyticsService }) => {
   });
 
   router.post('/page', (req, res) => {
-    const sessionId = path(['session', 'id'], req);
+    const sessionId = req.session.id;
+    const hostname = `${req.session.establishmentName}.${config.singleHostName}`;
```
Comment (andrewrlee) [inline server/routes/analytics.js:25]: now build up host name from derived establishment name and single hostname

File: test/test-helpers.js [hunk 4]
```diff
@@ -4,6 +4,8 @@ const nunjucksSetup = require('../server/utils/nunjucksSetup');
 function setupBasicApp(config = {}) {
   const app = express();
   app.set('view engine', 'html');
+  app.use(express.json());
+  app.use(express.urlencoded({ extended: true }));
```
Comment (andrewrlee) [inline test/test-helpers.js:8]: tests required parsing of json

**Review by jonhindle — APPROVED** [review]
LGTM

**Pull request merged**